server/controllers/chartsController/helpers.js

```diff
@@ -246,3 +243,44 @@ export const postProcessData = (data, studyTotals) => {
 
   return processedDataBySite
 }
+
+export const mongoQueryFromFilters = (filters) => {
+  if (!filters) {
+    return
+  }
+
+  const chrCritFilters = filters.chrcrit_part.filter((f) => f.value === true)
+  const includedExcludedFilters = filters.included_excluded.filter(
+    (f) => f.value === true
+  )
+
+  return [
+    {
+      chrcrit_part: {
+        $in: chrCritFilters.map(
+          (filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name]
+        ),
+      },
+    },
+    {
+      included_excluded: {
+        $in: includedExcludedFilters.map(
+          (filter) => FILTER_TO_MONGO_VALUE_MAP[filter.name]
+        ),
+      },
+    },
+  ]
+}
+
+export const sanitizeQueryParams = (filters) => {
+  return Object.keys(filters).reduce((sanitizedFilters, filter) => {
+    sanitizedFilters[filter] = filters[filter].map((filter) => {
+      return {
+        ...filter,
+        value: filter.value === 'true' ? true : false,
+      }
+    })
+
+    return sanitizedFilters
+  }, {})
```

Comment:
would this function be easier/not needed if we used 0 and 1 for the filter values instead of true and false? I'm wondering why we need this vs being able to use the query params directly.

Reply:
The Checkbox component uses the checked property to set the on or off state. The checked property only recognizes true or false as the appropriate values, additionally the qs library converts the boolean into a string.

Reply:
This is what the query params look like server side
`{
  chrcrit_part: [
    { name: 'HC', value: 'true' },
    { name: 'CHR', value: 'false' },
    { name: 'Missing', value: 'true' }
  ],
  included_excluded: [
    { name: 'Included', value: 'true' },
    { name: 'Excluded', value: 'false' },
    { name: 'Missing', value: 'true' }
  ]
}`

Reply:
Right, I believe the `Checkbox` component behaves similarly to an `input` of type `checkbox`. We don't need to use the `checked` attribute in setting the filter, though. We can use the `value` and we can set the `value` to whatever we want.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/248/commits/147bf8a9965b532f34bb176e1e651f339c9bf387